config.go

```diff
@@ -124,27 +85,53 @@ type SpanOption interface {
 	ApplySpan(*SpanConfig)
 }
 
+// NewEventConfig applies all the EventOptions to a returned SpanConfig. If no
+// timestamp option is passed, the returned SpanConfig will have a Timestamp
+// set to the call time, otherwise no validation is performed on the returned
+// SpanConfig.
+func NewEventConfig(options ...EventOption) *SpanConfig {
+	c := new(SpanConfig)
+	for _, option := range options {
+		option.ApplyEvent(c)
+	}
+	if c.Timestamp.IsZero() {
+		c.Timestamp = time.Now()
+	}
+	return c
+}
+
+// EventOption applies span life-cycle event options to a SpanConfig.
+type EventOption interface {
+	SpanOption
+
+	ApplyEvent(*SpanConfig)
+}
+
 type attributeSpanOption []label.KeyValue
 
-func (o attributeSpanOption) ApplySpan(c *SpanConfig) {
+func (o attributeSpanOption) ApplySpan(c *SpanConfig)  { o.apply(c) }
+func (o attributeSpanOption) ApplyEvent(c *SpanConfig) { o.apply(c) }
+func (o attributeSpanOption) apply(c *SpanConfig) {
 	c.Attributes = append(c.Attributes, []label.KeyValue(o)...)
 }
 
 // WithAttributes adds the attributes to a span. These attributes are meant to
 // provide additional information about the work the Span represents. The
 // attributes are added to the existing Span attributes, i.e. this does not
 // overwrite.
-func WithAttributes(attributes ...label.KeyValue) SpanOption {
+func WithAttributes(attributes ...label.KeyValue) EventOption {
```

Comment:
```suggestion
func WithAttributes(attributes ...label.KeyValue) attributeSpanOption {
```

Same question here about returning the concrete type.  How does that play with godoc?